src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -169,10 +169,12 @@ internal static void SetBrokerResult(Intent data, int resultCode)
                         };
                         break;
                     case (int)BrokerResponseCode.BrowserCodeError:
+                        dynamic errorResult = JObject.Parse(data.GetStringExtra(BrokerConstants.BrokerResultV2));
```

Comment:
Code needs to be more defensive: 

- make sure `data` and `data.GetStringExtra` are not null
- make sure `errorResult` has those keys that you look to. If they don't you still need to produce an "unknown error", setting Error and ErrorDescription to null is not a good user experience.

Reply:
data has a null check earlier in the code.
There are validation methods on the interactive and silent request that will catch the null value and replace it with the unknown error so using a null here is ok.